src/pages/locations/index.tsx

```diff
@@ -44,15 +46,50 @@ export const LocationsList: React.FC<IDataModelListProps> = (props) => {
     },
   });
 
-  const [viewMode, setViewMode] = useState<'table' | 'map'>('table');
+  const [viewMode, setViewMode] = useState<'table' | 'map'>(
+    props.viewMode ?? 'table',
+  );
 
+  const defaultHeight = props.height ?? '92%';
+  const isDisplayed = props.viewMode === undefined ? 'inline' : 'none';
   const toggleView = () => {
     setViewMode((prevMode) => (prevMode === 'table' ? 'map' : 'table'));
   };
 
+  const [searchQuery, setSearchQuery] = useState('');
+  const [filteredLocations, setFilteredLocations] = useState<Locations[]>([]);
+
+  const handleSearch = (query: string) => {
+    setSearchQuery(query);
+    const lowerCaseQuery = query.toLowerCase();
+
+    const filtered =
```

Comment:
Please Note: right now this may work because the dataSource will most likely give you all of the locations. In the future though, we will need to make this smarter so that it doesnt fetch all the locations and tries to display all of the locations on the map at once. Instead, the way it should work is that the map should identify the coordinates for the North East, North West, South East and South West, and send those values to the BE when fetching the locations, then we should only get and show the locations that are visible on the map.. when the user moves the map or zooms out, we would refetch. If we fetch all of the locations all at once, it can potentially overload the UI. For now though, I think we can just keep this in the back of our minds as something that we may need to optimize later.

Reply:
Technical debt noted